dotnet/src/SemanticKernel.Abstractions/AI/ChatCompletion/ChatCompletionServiceChatClient.cs

```diff
@@ -53,12 +53,12 @@ public ChatCompletionServiceChatClient(IChatCompletionService chatCompletionServ
     }
 
     /// <inheritdoc />
-    public async IAsyncEnumerable<ChatResponseUpdate> GetStreamingResponseAsync(IList<ChatMessage> chatMessages, ChatOptions? options = null, [EnumeratorCancellation] CancellationToken cancellationToken = default)
+    public async IAsyncEnumerable<ChatResponseUpdate> GetStreamingResponseAsync(IEnumerable<ChatMessage> messages, ChatOptions? options = null, [EnumeratorCancellation] CancellationToken cancellationToken = default)
     {
-        Verify.NotNull(chatMessages);
+        Verify.NotNull(messages);
 
         await foreach (var update in this._chatCompletionService.GetStreamingChatMessageContentsAsync(
-            new ChatHistory(chatMessages.Select(m => ChatCompletionServiceExtensions.ToChatMessageContent(m))),
+            new ChatHistory(messages.Select(m => ChatCompletionServiceExtensions.ToChatMessageContent(m))),
```

Comment:
Similarly here.

We'll need to track what's added to the history by the GetStreamingChatMessageContentsAsync call, and then after it completes and we've yielded everything else, create synthetic ChatResponseUpdate instances to represent each of the messages added to the chat history.

Unfortunately they'll be a bit out of order with regards to the other yielded content, but there's little we can do about that. It's inherent to the SK model.